modules/extreme_year.py

```diff
@@ -0,0 +1,117 @@
+"""
+this module will return the highest and lowest temperatures of a
+given year along with the most humid day.
+"""
+import glob
+
+
+def read_and_calculate_extremes(pattern, path):
+    """
+    this function reads all the files at the given path which contain the given pattern
+    from each file that matches the pattern finds the following:
+    1. max highest temperature
+    2. max lowest temperature
+    3. max humidity
+    :param pattern: a string which contains a year e.g '2006', '2007'
+    :param path: a string which contains path to a directory containing weather files
+    :return: a 3x2 list maximums where
+    maximums[0] = [max_highest_temperature, date]
+    maximums[1] = [max_lowest_temperature, date]
+    maximums[2] = [max_humidity, date]
+    date here is a string like: '2002-4-1', '2004-3-14'
+    """
+    files = glob.glob(path + f"*{pattern}*")
+    if not files:
+        print(
+            f"We don't have information regarding the "
+            f"year {pattern} in the given path"
+        )
+        return -1
+    # maximums[0] contains highest temperature and day
+    # maximums[1] contains lowest temperature and day
+    # maximums[2] contains most humidity and day
+    maximums = [[0, ""], [0, ""], [0, ""]]
+    initialized = False
+    # indexes to compare values with.
+    # highest temperature is stored at index 1 of weather reading
+    # lowest temperature is stored at index 3 of weather reading
+    # max humidity is stored at index 7 of weather reading
+    # see the weather files for further explanation
+    indexes = [1, 3, 7]
+    for i in files:
+        with open(i, "r") as file:
+            # skip first line as it contains field names
+            file.readline()
```

Comment:
Can you use `readlines()` instead , would not that make it a little simpler?

Reply:
i was using readlines() earlier but one drawback of readlines is that it reads all the file and keeps it in memory but we only need one row at a time therefore, i changed it into this. should i change it back to readlines() ?

Reply:
My idea was to use `readlines()` once store all(31 at max) the rows in a list and perform the looping on that. Your thought is good but in case we don't know the number of lines in a file when it could be 100 or 1000 etc. But in our case, we can only have 31 rows at max and I think we can keep them in memory and loop over them.

Reply:
done